source/geometry_geotiff_profile.h

```diff
@@ -0,0 +1,269 @@
+              tria1, {subdivisions_x_}, point_left_, point_right_);
+        } else if constexpr (dim == 2) {
+          dealii::GridGenerator::subdivided_hyper_rectangle(
+              tria1,
+              {subdivisions_x_, subdivisions_y_},
+              point_left_,
+              point_right_);
+        } else if constexpr (dim == 3) {
+          dealii::GridGenerator::subdivided_hyper_rectangle(
+              tria1,
+              {subdivisions_x_, subdivisions_y_, subdivisions_z_},
+              point_left_,
+              point_right_);
+        }
+
+        triangulation.copy_triangulation(tria1);
+        triangulation.reset_all_manifolds();
+        /* manifold id 0 for transfinite interpolation manifold */
+        triangulation.set_all_manifold_ids(0);
+
+        /* set boundary and manifold ids: */
+
+        for (auto cell : triangulation.active_cell_iterators()) {
+          for (auto f : cell->face_indices()) {
+            auto face = cell->face(f);
+            if (!face->at_boundary())
+              continue;
+            const auto position = face->center();
+
+            if (position[0] < point_left_[0] + 1.e-8) {
+              face->set_boundary_id(boundary_left_);
+              face->set_manifold_id(dealii::numbers::flat_manifold_id);
+            }
+            if (position[0] > point_right_[0] - 1.e-8) {
+              face->set_boundary_id(boundary_right_);
+              face->set_manifold_id(dealii::numbers::flat_manifold_id);
+            }
+
+            if constexpr (dim >= 2) {
+              if (position[1] < point_left_[1] + 1.e-8) {
+                face->set_boundary_id(boundary_bottom_);
+                /* manifold id 1 for ProfileManifold: */
+                face->set_manifold_id(1);
+              }
+              if (position[1] > point_right_[1] - 1.e-8) {
+                face->set_boundary_id(boundary_top_);
+                face->set_manifold_id(dealii::numbers::flat_manifold_id);
+              }
+            }
+
+            if constexpr (dim == 3) {
+              if (position[2] < point_left_[2] + 1.e-8) {
+                face->set_boundary_id(boundary_back_);
+                face->set_manifold_id(dealii::numbers::flat_manifold_id);
+              }
+              if (position[2] > point_right_[2] - 1.e-8) {
+                face->set_boundary_id(boundary_front_);
+                face->set_manifold_id(dealii::numbers::flat_manifold_id);
+              }
+            }
+          } /*for*/
+        }   /*for*/
+
+        const auto profile =
+            make_profile_manifold<dim>([&](dealii::Point<dim> point) {
+              /*
+               *
+               */
+              if constexpr (dim == 1) {
+                return 0.;
+              } else if constexpr (dim == 2) {
+                /*
+                 * Zero out the second coordinate when querying height
+                 * information in 2D.
+                 */
+                point[1] = 0;
```

Comment:
Should we add a parameter to allow the user to choose the y-plane value? i.e., `(x, y = constant, z)`?